config_test.go

```diff
@@ -0,0 +1,123 @@
+package main
+
+import (
+	"os"
+	"os/user"
+	"path/filepath"
+	"testing"
+)
+
+func init() {
+	// go test sets up the os.Args like so
+	// [/tmp/go-build689616692/b001/dcrdata.test -test.v=true
+	// -test.run=TestDefaultConfigAPIListen]
+	// however, the go-flags package then parses these and fails
+	// with unknown flags
+	// you will need to remove these flags before the flag parser is
+	// called like: os.Args = os.Args[:1]
+	os.Args = os.Args[:1]
```

Comment:
Having this in `_test.go` has it run with the `dcrdata.test`'s synthezied main, thus clobbering the testing args.  Notice you no longer have the stuff like `=== RUN   TestDefaultConfig
--- PASS: TestDefaultConfig (0.00s)` in the console.  You also get no output from `t.Logf` that you should get when running go test with `-v`.

I suggest to remove this `init` definition and instead using a `TestMain`:

```go
func TestMain(m *testing.M) {
	// Parse the -test.* flags before removing them from the command line
	// arguments list, which we do to allow go-flags to succeed.
	flag.Parse()
	os.Args = os.Args[:1]
	// Run the tests now that the testing package flags have been parsed.
	m.Run()
}
```